src/enumerable.cr

```diff
@@ -638,7 +638,7 @@ module Enumerable(T)
   # ```
   #
   # Returns `nil` if *obj* is not in the collection.
-  def index(obj)
+  def index(obj) : Int32?
```

Comment:
#10582 added a return type restriction `Int32?` to the override in `Array#index` (https://github.com/crystal-lang/crystal/pull/10582/files#diff-a12e3cba63cd83b9e746cd9eab9d85f5ddebd05b696133cd494aa83174a48d16R2252). This method delegates to `super`, so it can only hold if `Enumerable#index` returns `Int32?` as well. It's either both or none. Adding the return type restriction to `Enumerable` doesn't seem like an issue, so I think it's best to just add the restriction there, too (instead of removing it on `Array#index`).